test.disabled/ejabberd_tests/tests/mam_SUITE.erl

```diff
@@ -1354,6 +1377,42 @@ archived(Config) ->
     %% Made fresh in init_per_testcase
     escalus:story(Config, [{alice, 1}, {bob, 1}], F).
 
+
+message_with_stanzaid_and_archived(Config) ->
+    F = fun(Alice, Bob) ->
+        %% Archive must be empty.
+        %% Alice sends "OH, HAI!" to Bob.
+        escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"OH, HAI!">>)),
+
+        %% Bob receives a message.
+        Msg = escalus:wait_for_stanza(Bob),
+        try
+        Arc_archived = exml_query:subelement(Msg, <<"archived">>),
+	Arc_stanzaid = exml_query:subelement(Msg, <<"stanza-id">>),
+        %% JID of the archive (i.e. where the client would send queries to)
+        By_archived  = exml_query:attr(Arc_archived, <<"by">>),
+        By_stanzaid  = exml_query:attr(Arc_stanzaid, <<"by">>),
+        %% Attribute giving the message's UID within the archive.
+        Id_archived  = exml_query:attr(Arc_archived, <<"id">>),
+        Id_stanzaid  = exml_query:attr(Arc_stanzaid, <<"id">>),
+
+	%% The attributes of <stanza-id/> and <archived/> are the same
+	?assert_equal(By_archived, By_stanzaid),
+	?assert_equal(Id_archived, Id_stanzaid),
+
+	%% stanza-id has a namespace 'urn:xmpp:sid:0'
+	Xmlns = exml_query:attr(Arc_stanzaid, <<"xmlns">>),
```

Comment:
What you wrote is correct but `<<"urn:xmpp:sid:0">> = exml_query:attr(Arc_stanzaid, <<"xmlns">>)` would work equally well and is shorter.